lib/adapters/definition-adapter.ts

```diff
@@ -68,11 +68,15 @@ export default class DefinitionAdapter {
    * @param locationResult Either a single {Location} object or an {Array} of {Locations}.
    * @returns An {Array} of {Location}s or {null} if the locationResult was null.
    */
-  public static normalizeLocations(locationResult: Location | Location[]): Location[] | null {
+  public static normalizeLocations(
+    locationResult: Location | (Location | LocationLink)[]
+  ): (Location | LocationLink)[] | null {
```

Comment:
Based on the docs, this is either `Location` or `LocationLink`. It is not a mixed array.
```suggestion
    locationResult: Location | Location[] | LocationLink[]
  ): Location[] | LocationLink[] | null {
```

Reply:
Thank you for review.
When I make this change, I get the following error:
```ts
lib/adapters/definition-adapter.ts:77:80 - error TS2349: This expression is not callable.
  Each member of the union type '{ <S extends Location>(predicate: (value: Location, index: number, array: Location[]) => value is S, thisArg?: any): S[]; (predicate: (value: Location, index: number, array: Location[]) => unknown, thisArg?: any): Location[]; } | { ...; }' has signatures, but none of those signatures are compatible with each other.

77     return (Array.isArray(locationResult) ? locationResult : [locationResult]).filter(
                                                                                  ~~~~~~

lib/adapters/definition-adapter.ts:78:8 - error TS7006: Parameter 'd' implicitly has an 'any' type.

78       (d) => ("range" in d ? d.range : d.targetRange).start != null
          ~
```

For example, is it okay to add  `as any []`  as shown below?

```ts
  public static normalizeLocations(
    locationResult: Location | Location[] | LocationLink[] | null
  ): Location[] | LocationLink[] | null {
    if (locationResult == null) {
      return null
    }
    return (Array.isArray(locationResult) ? locationResult as any[] : [locationResult]).filter(
                                                           ^^^^^^^^^
      (d) => ("range" in d ? d.range : d.targetRange).start != null
    )
  }
```

Reply:
I think you should add a type to `d`. Something like:

```js
(d: Location | LocationLink) => ("range" in d ? d.range : d.targetRange).start != null
```

Reply:
Fixed in the way you said.